#7469 - OAI-PMH API key authentication and embedding of digital objects

mcantelon opened the pull request:
Added OAI-PMH API key authentication and embedding of digital objects, as Atom XML, in record XML.

mcantelon committed:
Embed digital objects into OAI using Atom feed.

mcantelon committed:
Generate OAI API keys via user admin. refs #7469

mcantelon committed:
Fixed issue with enabling OAI plugin. refs #7469

mcantelon committed:
Process request only with OAI API key. refs #7469

Added a setting to disable OAI API key authentication

Added logic to require OAI requests provide key if an OAI
key exists and OAI key authentication is enabled. Sign in
user if correct OAI API key supplied so ACL checks in
template work.

mcantelon committed:
Include about tag in OAI list output. refs #7468

Created a separate template for the about tag and made it so
the about tag is included in OAI-PHP output for the listRecords
verb (as well as getRecord).

mcantelon committed:
Replaced primary key with slug. refs #7468

Replaced the use of a digital object's primary key with its
slug.

mcantelon committed:
Allow OAI API auth if not required. refs #7469

If OAI API auth isn't required, allow requests to be made using
OAI API auth keys to get additional data (reflecting permissions
of user with matching auth key).

sevein commented:
Nicely done! My only rant is that the oai_enabled QubitSetting makes the code harder to read. Is the setting object null? Then, we change the type of value to bool and... is true? is false? I'm not sure if you have any time left for this, but isPluginEnabled() would be a nice thing to see!

mcantelon committed:
Changes to match code standards.

mcantelon committed:
Use helpers for 403 response.

mcantelon committed:
Got rid of redundant OAI setting. refs #7469

Removed the "oai_enabled" setting as we can just test to see if the
plugin is enabled. Simplified some code.

mcantelon committed:
Removed OAI disabled page. refs #7469

Removed page that's no longer needed.

mcantelon commented:
Merged.
